app/lib/actions/pulp/repository/create.rb

```diff
@@ -60,6 +62,8 @@ def yum_or_iso_importer
           importer.ssl_validation  = input[:ssl_validation]
           importer.download_policy = input[:download_policy] if input[:content_type] == ::Katello::Repository::YUM_TYPE
           importer.remove_missing  = input[:mirror_on_sync] if input[:content_type] == ::Katello::Repository::YUM_TYPE
+          importer.basic_auth_username = upstream_username.blank? ? '' : upstream_username
+          importer.basic_auth_password = upstream_password.blank? ? '' : upstream_password
```

Comment:
The username/password need to be added to the input in the call to plan_action() on line 16. Here and the other importers below should reference input[:upstream_username] etc.